src/transformers/modeling_utils.py

```diff
@@ -3460,6 +3460,42 @@ def reverse_bettertransformer(self):
 
         return BetterTransformer.reverse(self)
 
+    def warn_if_padding_and_no_attention_mask(self, input_ids, attention_mask):
+        """
+        Shows a one-time warning if the input_ids appears to contain padding and no attention mask was given.
+        """
+        if (attention_mask is not None) or (self.config.pad_token_id is None):
+            return
+
+        if not hasattr(self, "warnings_issued"):
+            self.warnings_issued = {}
+        if self.warnings_issued.get("padding_and_no_attention_mask", False):
+            return
+
+        # Check only the first and last input IDs to reduce overhead.
+        if self.config.pad_token_id in input_ids[:, -1] or self.config.pad_token_id in input_ids[:, 1]:
```

Comment:
I took the liberty to add checks just for the start/end of the input_ids to reduce overhead, what do you think of this?

My understanding is that, in most cases, the padding should only show up in the beginning or the end, and that this should catch many of the mistakes.